### ISSUE TITLE
suffix "_value" is added to out metric name from prometheus client v2/collector

### ISSUE THREAD

**Issue by Logic0** [influxdata/telegraf#7171]
### Telegraf version
Telegraf 1.13.4 (git: HEAD ffabd6b5)

### Relevant telegraf.conf:
```toml
[[inputs.statsd]]
#   ## Protocol, must be "tcp", "udp", "udp4" or "udp6" (default=udp)
    protocol = "udp"
#
#   ## MaxTCPConnection - applicable when protocol is set to tcp (default=250)
#   max_tcp_connections = 250
#
#   ## Enable TCP keep alive probes (default=false)
#   tcp_keep_alive = false
#
#   ## Specifies the keep-alive period for an active network connection.
#   ## Only applies to TCP sockets and will be ignored if tcp_keep_alive is false.
#   ## Defaults to the OS configuration.
#   # tcp_keep_alive_period = "2h"
#
#   ## Address and port to host UDP listener on
    service_address = ":8125"
#
#   ## The following configuration options control when telegraf clears it's cache
#   ## of previous values. If set to false, then telegraf will only clear it's
#   ## cache when the daemon is restarted.
#   ## Reset gauges every interval (default=true)
    delete_gauges = false
#   ## Reset counters every interval (default=true)
    delete_counters = false
#   ## Reset sets every interval (default=true)
    delete_sets = true
#   ## Reset timings & histograms every interval (default=true)
    delete_timings = true
#
#   ## Percentiles to calculate for timing & histogram stats
#   percentiles = [50.0, 90.0, 99.0, 99.9, 99.95, 100.0]
#
#   ## separator to use between elements of a statsd metric
    metric_separator = "_"
#
#   ## Parses tags in the datadog statsd format
#   ## http://docs.datadoghq.com/guides/dogstatsd/
#   parse_data_dog_tags = false
#
#   ## Parses datadog extensions to the statsd format
#   datadog_extensions = false
#
#   ## Statsd data translation templates, more info can be read here:
#   ## https://github.com/influxdata/telegraf/blob/master/docs/TEMPLATE_PATTERN.md
#   # templates = [
#   #     "cpu.* measurement*"
#   # ]
#
#   ## Number of UDP messages allowed to queue up, once filled,
#   ## the statsd server will start dropping packets
#   allowed_pending_messages = 10000
#
#   ## Number of timing/histogram values to track per-measurement in the
#   ## calculation of percentiles. Raising this limit increases the accuracy
#   ## of percentiles but also increases the memory usage and cpu time.
#   percentile_limit = 1000

[[outputs.prometheus_client]]
    ## Address to listen on
    listen = ":9273"
#
#   ## Metric version controls the mapping from Telegraf metrics into
#   ## Prometheus format.  When using the prometheus input, use the same value in
#   ## both plugins to ensure metrics are round-tripped without modification.
#   ##
#   ##   example: metric_version = 1; deprecated in 1.13
#   ##            metric_version = 2; recommended version
    metric_version = 2
#
#   ## Use HTTP Basic Authentication.
#   # basic_username = "Foo"
#   # basic_password = "Bar"
#
#   ## If set, the IP Ranges which are allowed to access metrics.
#   ##   ex: ip_range = ["192.168.0.0/24", "192.168.1.0/30"]
#   # ip_range = []
#
#   ## Path to publish the metrics on.
    path = "/metrics"
#
#   ## Expiration interval for each metric. 0 == no expiration
    expiration_interval = "30d"  # 60s
#
#   ## Collectors to enable, valid entries are "gocollector" and "process".
#   ## If unset, both are enabled.
    collectors_exclude = ["gocollector", "process"]
#
#   ## Send string metrics as Prometheus labels.
#   ## Unless set to false all string metrics will be sent as labels.
#   # string_as_label = true
#
#   ## If set, enable TLS with the given certificate.
#   # tls_cert = "/etc/ssl/telegraf.crt"
#   # tls_key = "/etc/ssl/telegraf.key"
#
#   ## Set one or more allowed client CA certificate file names to
#   ## enable mutually authenticated TLS connections
#   # tls_allowed_cacerts = ["/etc/telegraf/clientca.pem"]
#
#   ## Export metric collection time.
    export_timestamp = true

```

### System info:

Linux VM_48_5_centos 3.10.0-957.27.2.el7.x86_64 #1 SMP Mon Jul 29 17:46:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

### Steps to reproduce:
```shell
## send a counter metric foo2 to statsd input plugin
# echo "foo2:1|c" | nc -u -w1 127.0.0.1  8125
```
with v2/collector (set outputs.prometheus_client.metric_version = 2)
```shell
# curl http://127.0.0.1:9273/metrics
# HELP foo2_value Telegraf collected metric
# TYPE foo2_value counter
foo2_value{metric_type="counter"} 1
```
with v1/collector (set outputs.prometheus_client.metric_version = 1)
```shell
# curl http://127.0.0.1:9273/metrics
# HELP foo2 Telegraf collected metric
# TYPE foo2 counter
foo2{metric_type="counter"} 1
```

### Expected behavior:

out metric name is same in v1 and v2 collector

### Actual behavior:

foo2_value in v2 and foo2 in v1.

### Additional info:
from the plugins/outputs/promethues_client/v1/collector.go source, filed name "value" is processed specially:
```Go
func (c *Collector) Add(metrics []telegraf.Metric) error {
    ...
    if mname == "" {
        // special procedure for field name "value"
        if fn == "value" {
            mname = sanitize(point.Name())
        } else {
            mname = sanitize(fmt.Sprintf("%s_%s", point.Name(), fn))
        }
    }
```
from v2 metric name generator serializers/prometheus/convert.go source:
```Go
// MetricName returns the Prometheus metric name.
func MetricName(measurement, fieldKey string, valueType telegraf.ValueType) string {
	switch valueType {
	case telegraf.Histogram, telegraf.Summary:
		switch {
		case strings.HasSuffix(fieldKey, "_bucket"):
			fieldKey = strings.TrimSuffix(fieldKey, "_bucket")
		case strings.HasSuffix(fieldKey, "_sum"):
			fieldKey = strings.TrimSuffix(fieldKey, "_sum")
		case strings.HasSuffix(fieldKey, "_count"):
			fieldKey = strings.TrimSuffix(fieldKey, "_count")
		}
	}

	if measurement == "prometheus" {
		return fieldKey
	}
        // every filed name were attached to metric name as a suffix, including "value"
	return measurement + "_" + fieldKey
}
```

filed key "value"  is the default field name used in statsd input plugin to store the value.
source plugins/inputs/statsd.go:
```Go

defaultFieldName = "value"

// parseName parses the given bucket name with the list of bucket maps in the
// config file. If there is a match, it will parse the name of the metric and
// map of tags.
// Return values are (<name>, <field>, <tags>)
func (s *Statsd) parseName(bucket string) (string, string, map[string]string) {
        ...
	if field == "" {
		field = defaultFieldName
	}
	return name, field, tags
```

**Comment by danielnelson** [comment]
> Expected behavior: out metric name is same in v1 and v2 collector

This wasn't my intention with `metric_version = 2`, in fact the whole point was for it to produce different results.  However, this is obviously not ideal output and I think we likely ought to revisit the statsd input and update how it produces metrics as well (similar to what we did for the prometheus input).

There is a workaround, but it only seems to work if your actual stats always have at least one dot in them,  use a template to set the field name:
```
[[inputs.statsd]]
  template = ["measurement.field*"]
```
Sending `foo.bar:1|c` will produce `foo_bar{metric_type="counter"} 1` or in line protocol:
```
foo,metric_type=counter bar=1i
```

For some reason this doesn't work if the stats have no dots, `template = ["field*"]`doesn't do what I expect, this is probably a bug.

I think when 1.14 is released, you could use the template processor followed by the pivot processor to set the field name to the measurement, but this seems overly complicated.

**Comment by danielnelson** [comment]
Before closing this issue, I'd like to fix the statsd template to support capturing a field name only:
```
template = ["field*"]
```

This would use the plugin default measurement name, turning `foo:1|c` into:
```
statsd,metric_type=counter foo=1i
```

**Comment by Logic0** [comment]
Much appriciated.  
Saved me out of a custom version maintenance. ^_^

**Comment by AntonioAlfrz** [comment]
We have experienced the same behavior with exec input plugin and _data format_:

```toml
[[inputs.exec]]
  commands = ["/bin/bash -c \"curl -s http://127.0.0.1:8080/tunnelcount | grep -Eo '[0-9]+'\""]
  data_format = "value"
  name_override = "tunnels_total"

# OUTPUTS

[[outputs.prometheus_client]]
  listen = ":9126"
  expiration_interval = "120s"
  collectors_exclude = ["gocollector", "process"]
  metric_version = 2
```
```bash
# It shows tunnels_total as metric name
telegraf --test --input-filter exec
```
However, the metric name in the endpoint is **tunnels_total_value**

Thank you

**Comment by danielnelson** [comment]
@AntonioAlfrz For your issue we should so something similar for `data_format = "value"` to what I proposed above as the fix for statsd.  Instead of only allowing the field key to be set to `value`, the user of the value data format should be able to set the field key to something that describes what the value is.  This will allow you to do something like this and have the desired output from the prometheus output plugin.
```
[[inputs.exec]]
  commands = ["/bin/bash -c \"curl -s http://127.0.0.1:8080/tunnelcount | grep -Eo '[0-9]+'\""]
  data_format = "value"
  value_field = "total"
  name_override = "tunnels"
```

I opened #7702 to track this.
